pkg/utils/image.go

```diff
@@ -25,15 +25,27 @@ import (
 // It will give preference to images that are set in the etcd spec and only if the image is not found in it should
 // it be picked up from the image vector if it's set there.
 // A return value of nil for either of the images indicates that the image is not set.
-func GetEtcdImages(etcd *druidv1alpha1.Etcd, iv imagevector.ImageVector) (*string, *string, error) {
-	etcdImage, err := chooseImage(common.Etcd, etcd.Spec.Etcd.Image, iv)
+func GetEtcdImages(etcd *druidv1alpha1.Etcd, iv imagevector.ImageVector, useEtcdWrapper bool) (*string, *string, error) {
+	var err error
+	var etcdImage, etcdBackupRestoreImage *string
```

Comment:
A more cleaner way to do it would be:
```go
func getEtcdImageKeys(useEtcdWrapper bool) (etcdImageKey string, etcdbrImageKey string) {
	switch useEtcdWrapper {
	case true:
		etcdImageKey = common.EtcdWrapper
		etcdbrImageKey = common.BackRestoreDistroless
	default:
		etcdImageKey = common.Etcd
		etcdbrImageKey = common.BackupRestore
	}
	return
}

func GetEtcdImages(etcd *druidv1alpha1.Etcd, iv imagevector.ImageVector, useEtcdWrapper bool) (*string, *string, error) {
	etcdImageKey, etcdbrImageKey := getEtcdImageKeys(useEtcdWrapper)
	etcdImage, err := chooseImage(etcdImageKey, etcd.Spec.Etcd.Image, iv)
	if err != nil {
		return nil, nil, err
	}
	etcdBackupRestoreImage, err := chooseImage(etcdbrImageKey, etcd.Spec.Backup.Image, iv)
	if err != nil {
		return nil, nil, err
	}
	return etcdImage, etcdBackupRestoreImage, nil
}
```